packages/manager/apps/pci-kubernetes/src/pages/new/hooks/useEstimationViewModel.ts

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react';
+
+import { NodePoolPrice, TClusterPlan, TClusterPlanEnum, TPlan } from '@/types';
+
+export type EstimationPriceViewModel = {
+  label: string;
+  value?: string;
+  show: boolean;
+};
+
+const useEstimationModel =
+  (
+    t: (translationKey: string) => string,
+    plans: TPlan[],
+    getFormattedMonthlyCatalogPrice: (price: number) => string,
+    convertHourlyPriceToMonthly: (price: number) => number,
+  ) =>
+  (
+    plan: TClusterPlan,
+    nodePools?: NodePoolPrice[] | null,
+    options?: { has3AZ?: boolean; showSavingPlan?: boolean },
+  ): EstimationPriceViewModel[] => {
```

Comment:
Currification is great here and is much more adapted to dependencies injection.

But, you need to split dependencies belonging to the infra part (`t, getFormattedMonthlyCatalogPrice, convertHourlyPriceToMonthly`) and model properties so that you could be as close as possible of a [Reader](https://hackage.haskell.org/package/mtl-2.3.1/docs/Control-Monad-Reader.html) monad.

And then you'll have : 

```ts
const selectEstimationPriceFromPlans = (deps: Deps) => (model: Model) => {
 // your logic here
}
```

that can be called with a bit of curry 🌶️

```ts
const estimation = selectEstimationPriceFromPlans(deps)(model)
```